### PR TITLE
[Project Addition] Tiger Identification using DL

### PR DESCRIPTION
**Pull request by jeet-Abhi123**
# Pull Request for DL-Simplified 💡

#248  :  Tiger Identification using DL 

- **Info about the related issue (Aim of the project)** :  The aim of this project is to develop and evaluate deep learning models for accurately re-identifying individual tigers from images.
- **Name:** Abhijeet Kaithwas
- **GitHub ID:**  https://github.com/jeet-Abhi123 
- **Email ID:** abhijeet.kaithwas13@gmail.com 
- **Idenitfy yourself: (Mention in which program you are contributing in. **  GSSOC'24 Contributor

Closes: #248 

### Describe the add-ons or changes you've made 📃

This project utilizes various deep learning models to carry out the tiger Re-ID task. The models employed are DenseNet, VGG16, ResNet50, and EfficientNet. The main goal is to develop and assess these models using a designated training set, followed by testing their effectiveness on a different test set. In the testing phase, each image acts as a query image, while the rest of the images in the test set function as the "gallery" or "database." The outcome for each query is anticipated to be a ranked list of gallery images, highlighting the most probable matches.

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Describe how it has been tested
I have tested the implementation using a separate test dataset and verified the classification accuracy of each model. Additionally, I have compared the results with the expected outcomes to ensure accuracy.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Please follow the project structure and arrange your files according to that.

@jeet-Abhi123

**Comment by jeet-Abhi123** [comment]
@abhisheks008  where is the project structure given?

**Comment by jeet-Abhi123** [comment]
@abhisheks008  Please take a look now.

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
1. Merge all the `.ipynb` files into a signle file and upload a single Jupyter Notebook file for the project.
2. Follow the README template, https://github.com/abhisheks008/DL-Simplified/blob/main/.github/readme_template.md
3. The file name should be `README.md` not `readme.md`.
4. In the Dataset folder update the README file name and put a brief about the dataset instead of just putting the dataset source.

@jeet-Abhi123

**Comment by jeet-Abhi123** [comment]
> 1. Merge all the `.ipynb` files into a signle file and upload a single Jupyter Notebook file for the project.
> 2. Follow the README template, https://github.com/abhisheks008/DL-Simplified/blob/main/.github/readme_template.md
> 3. The file name should be `README.md` not `readme.md`.
> 4. In the Dataset folder update the README file name and put a brief about the dataset instead of just putting the dataset source.
> 
> @jeet-Abhi123

@abhisheks008   Is it compulsory to merge all .ipyn files into one because I have seen different projects in your repository which have different models and they are present in different .ipyn files. 
In case if you didn't like the above statement, merging all the .ipyn and running them again will take a lot of time because each model was taking 40 mins of training.

**Review by abhisheks008 — APPROVED** [review]
Approved @jeet-Abhi123
